File: chat-react-native/ios/Providers/RNKustomerPushProvider.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE (RNKustomerPushProvider, NSObject)

RCT_EXTERN_METHOD(setToken : (NSString *)token)
RCT_EXTERN_METHOD(clearToken
                  : (RCTPromiseResolveBlock)resolve rejecter
                  : (RCTPromiseRejectBlock)reject)
@end

File: chat-react-native/ios/Providers/RNKustomerPushProvider.swift
import KustomerChat
import UIKit

@objcMembers
@objc(RNKustomerPushProvider)
class RNKustomerPushProvider: NSObject {
    static func requiresMainQueueSetup() -> Bool {
      return true
    }
    
    func setToken(token: String) {
        let data = token.data(using: .utf8)!
        KustomerClient.shared.pushProvider.didRegisterForRemoteNotifications(deviceToken: data)
    }

    func clearToken(_ resolve: @escaping RCTPromiseResolve<Void?>, rejecter reject: @escaping RCTPromiseRejectBlock) {
        KustomerClient.shared.pushProvider.deregisterCurrentDeviceForPushNotifications { result in
            switch result {
            case .success:
                resolve(nil)
            case .failure(let error):
                reject(error.localizedDescription, nil, error)
            }
        }
    }
}
